wagtail/contrib/postgres_search/utils.py

```diff
@@ -68,15 +66,18 @@ def get_descendants_content_types_pks(models, db_alias):
               for descendant_model in get_descendant_models(model)), db_alias)
 
 
-@lru_cache()
 def get_content_types_pks(models, db_alias):
     # We import it locally because this file is loaded before apps are ready.
     from django.contrib.contenttypes.models import ContentType
-    return list(ContentType._default_manager.using(db_alias)
-                .filter(OR([Q(app_label=model._meta.app_label,
-                              model=model._meta.model_name)
-                            for model in models]))
-                .values_list('pk', flat=True))
+    ct_manager = ContentType.objects
+    original_db_alias = ct_manager.db
+    try:
+        ct_manager._db = db_alias
+        content_types_dict = ContentType.objects.get_for_models(*models)
+    finally:
+        ct_manager._db = original_db_alias
```

Comment:
Actually, after further thinking I was not really happy with this solution as it was not thread-safe: the manager object stays the same across threads, so there’s a extremely tiny chance that modifying `.db` can affect another thread.

After a small search, I found an undocumented way to copy a manager for another database: `Manager.db_manager(db_alias)`. It’s roughly the equivalent of `Queryset.using(db_alias)` for `Manager`. Even though that’s undocumented, it is used multiple times in Django tests, and it will explicitely raise an `AttributeError` if that API changes, while setting `.db` would probably not.